devices/common_patches/h7_ethernet.yaml

```diff
@@ -0,0 +1,2082 @@
+            description: Auxiliary Snapshot 2 Enable
+            bitOffset: 6 
+            bitWidth: 1
+          ATSEN1:
+            description: Auxiliary Snapshot 1 Enable
+            bitOffset: 5 
+            bitWidth: 1
+          ATSEN0:
+            description: Auxiliary Snapshot 0 Enable
+            bitOffset: 4 
+            bitWidth: 1
+          ATSFC:
+            description: Auxiliary Snapshot FIFO Clear
+            bitOffset: 0 
+            bitWidth: 1
+      MACATSNR:
+        description: Auxiliary timestamp nanoseconds register
+        addressOffset: 0x0B48
+        resetValue: 0x00000000
+        access: read-only
+        fields:
+          AUXTSLO:
+            description: Auxiliary Timestamp
+            bitOffset: 0 
+            bitWidth: 31
+      MACATSSR:
+        description: Auxiliary timestamp seconds register
+        addressOffset: 0x0B4C
+        resetValue: 0x00000000
+        access: read-only
+        fields:
+          AUXTSHI:
+            description: Auxiliary Timestamp
+            bitOffset: 0 
+            bitWidth: 32
+      MACTSIACR:
+        description: Timestamp Ingress asymmetric correction register
+        addressOffset: 0x0B50
+        resetValue: 0x00000000
+        access: read-write
+        fields:
+          OSTIAC:
+            description: One-Step Timestamp Ingress Asymmetry Correction
+            bitOffset: 0 
+            bitWidth: 32
+      MACTSEACR:
+        description: Timestamp Egress asymmetric correction register
+        addressOffset: 0x0B54
+        resetValue: 0x00000000
+        access: read-write
+        fields:
+          OSTEAC:
+            description: One-Step Timestamp Egress Asymmetry Correction
+            bitOffset: 0 
+            bitWidth: 32
+      MACTSICNR:
+        description: Timestamp Ingress correction nanosecond register
+        addressOffset: 0x0B58
+        resetValue: 0x00000000
+        access: read-write
+        fields:
+          TSIC:
+            description: Timestamp Ingress Correction
+            bitOffset: 0 
+            bitWidth: 32
+      MACTSECNR:
+        description: Timestamp Egress correction nanosecond register
+        addressOffset: 0x0B5C
+        resetValue: 0x00000000
+        access: read-write
+        fields:
+          TSEC:
+            description: Timestamp Egress Correction
+            bitOffset: 0 
+            bitWidth: 32
+      MACPPSCR:
+        description: PPS control register
+        addressOffset: 0x0B70
+        resetValue: 0x00000000
+        access: read-write
+        fields:
+          TRGTMODSEL0:
+            description: Target Time Register Mode for PPS Output
+            bitOffset: 5 
+            bitWidth: 2
+          PPSEN0:
+            description: Flexible PPS Output Mode Enable
+            bitOffset: 4 
+            bitWidth: 1
+          PPSCTRL:
+            description: PPS Output Frequency Control
+            bitOffset: 0 
+            bitWidth: 4
+      MACPPSCR:
```

Comment:
I don't understand what's going on here and briefly thought I was going mad. The reference manual lists `MACPPSCR` twice, both at the same offset, with the same `TRGTMODSEL0` and `PPSEN0` but totally different `PPSCTRL`/`PPSCMD` in the bottom 4 bits. We can't just list both registers identically with the same offset but who knows what the correct field is. Any ideas?

Reply:
This had me super confused at first after you pointed it out - it looks like what's actually going on is that PPSEN0 modifies the behavior of bits [0:3] - when PPSEN0 is 0, bits 0:3 act as PPSCTRL, but when PPSEN0 is 1, bits 0:3 act as PPSCMD.

That being said, I have no idea what the right course for handling this in the crate is.

Reply:
Looks like some Intel/Altera parts use the same (or similar) Synopsys IP for their ethernet peripherals, and their docs are a little clearer on this point: https://www.intel.co.jp/content/dam/altera-www/global/ja_JP/pdfs/literature/hb/cyclone-v/hps.pdf